pkg/controller/installation/products/fuse/reconciler.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	v1alpha12 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	syn "github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	errors2 "k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	defaultInstallationNamespace = "fuse"
+	defaultSubscriptionName      = "syndesis"
+)
+
+type Reconciler struct {
+	client        pkgclient.Client
+	coreClient    *kubernetes.Clientset
+	Config        *config.Fuse
+	ConfigManager config.ConfigReadWriter
+	mpm           marketplace.MarketplaceInterface
+	logger        *logrus.Entry
+	nsReconciler  *resources.NamespaceReconciler
+}
+
+func NewReconciler(client pkgclient.Client, rc *rest.Config, coreClient *kubernetes.Clientset, configManager config.ConfigReadWriter, instance *v1alpha1.Installation, logger *logrus.Entry) (*Reconciler, error) {
+	fuseConfig, err := configManager.ReadFuse()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve keycloak codeReadyConfig")
+	}
+
+	if fuseConfig.GetNamespace() == "" {
+		fuseConfig.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+	if err = fuseConfig.Validate(); err != nil {
+		return nil, errors.Wrap(err, "keycloak codeReadyConfig is not valid")
+	}
+
+	mpm := marketplace.NewManager(client, rc)
+	nsr := resources.NewNamespaceReconciler(client, logger)
+	return &Reconciler{client: client,
+		coreClient:    coreClient,
+		ConfigManager: configManager,
+		Config:        fuseConfig,
+		mpm:           mpm,
+		logger:        logger,
+		nsReconciler:  nsr,
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(inst *v1alpha1.Installation) (v1alpha1.StatusPhase, error) {
+	ctx := context.TODO()
+
+	reconciledPhase, err := r.reconcileNamespace(ctx, inst)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile namespace for fuse ")
+	}
+
+	reconciledPhase, err = r.reconcileSubscription(ctx)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile subscription for fuse ")
+	}
+
+	reconciledPhase, err = r.handleAwaitingOperator(ctx)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile subscription for fuse ")
+	}
+	reconciledPhase, err = r.reconcileCustomResource(ctx, inst)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, " failed to reconcile fuse custom resource ")
+	}
+
+	logrus.Info("End of reconcile Phase : ", reconciledPhase)
+
+	return v1alpha1.StatusPhase(v1alpha1.PhaseCompleted), nil
+}
+
+func (r *Reconciler) reconcileNamespace(ctx context.Context, inst *v1alpha1.Installation) (v1alpha1.StatusPhase, error) {
+	if !inst.Spec.CreateNamespaces {
```

Comment:
I think this can be removed since the namespace wouldn't be created if it existed

Reply:
yep. Removed now